assembly.py

```diff
@@ -339,6 +340,49 @@ def parser_assemble_trinity(parser=argparse.ArgumentParser()):
 __commands__.append(('assemble_trinity', parser_assemble_trinity))
 
 
+def assemble_spades(
+    inBam,
+    outFasta,
+    spades_opts='',
+    always_succeed=False,
+    threads=1,
+):
+    ''' This step runs the SPAdes assembler.
+    '''
+
+    with tools.picard.SamToFastqTool().execute_tmp(inBam, includeUnpaired=True) as (reads_fwd, reads_bwd, reads_unpaired):
+        try:
+            tools.spades.SpadesTool().assemble(reads_fwd=reads_fwd, reads_bwd=reads_bwd, reads_unpaired=reads_unpaired,
+                                               contigs_out=outFasta, spades_opts=spades_opts)
+            if not os.path.isfile(outFasta) or os.path.getsize(outFasta) == 0:
```

Comment:
Differentiate how?  My understanding was that always_succeed means output empty contigs in either case, and "not always_succeed" means fail in either case.  (Should probably also remove contigs shorter than a given length before checking for emptiness?)

always_succeed is checked on the 'except' branch before raising an exception.  The raise statement on the try branch will move to the except branch where always_succeed is checked.